test/controllers/test_case_controller.py

```diff
@@ -21,6 +23,12 @@ def test_get_all_business_ids_for_collection_exercise(self, mock_engine):
         test_output = case_controller.get_all_business_ids_for_collection_exercise(EXERCISE_ID)
         self.assertEqual(expected_output, test_output)
 
+    def test_get_all_business_ids_for_collection_exercise_sqlalchemyerror(self):
+        with self.assertRaises(SQLAlchemyError):
+            expected_output = None
+            test_output = case_controller.get_all_business_ids_for_collection_exercise("bad_exercise_id")
+            self.assertEqual(expected_output, test_output)
+
```

Comment:
While yes this is fine, I was referring to how you raise the error, this change is just to how you assert the error. Rather than having it fall over by inputting bad data (line 29 in the above snippet) you can patch the method that would be returning the SQLAlchemy error and have it do `method_mock..side_effect = SQLAlchemyError()` instead of having it raise the error through incorrect input data because that may raise a different error or may never be passed to this method because of data sanitisation for example

Reply:
Mocks and side effects have been added to the newly-introduced unit tests.